### ISSUE TITLE
Rails cannot bundle with coffee-rails

### ISSUE THREAD

**Issue by RasPat1** [rails/rails#29202]
### Steps to reproduce
Trying to run a bundle on rails 5.2.0-alpha will give a dependency error on railties version.

This is a resolved issue in coffee-rails as of 2 months ago. [Issue#88](https://github.com/rails/coffee-rails/issues/88)
However it has not been released yet.
Merged on [March 23](https://github.com/rails/coffee-rails/commit/608c708298afdc62c9de45d96b28b7db558aebee)
Last release cut 11 months ago on [June 30, 2016](https://github.com/rails/coffee-rails/commit/b3d64eb497db07346748a0504e353741b621bc23)

### Open Questions
What is the typical way to handle when alpha development requires unreleased versions of a dependency? 
In general, Coffee-rails has no need to release the gem for public use since only alpha requires it.
Should alpha versions always use the latest github master version?
Should we notify coffee-rails to push a new release?
Why has it taken 2 months to release an unbreaking fix?

### Proposed Fix
Propose updating gemfile to resolve to coffee-rails git repo.

### Expected behavior
Rails should be able to bundle

### Actual behavior
Rails was unable to bundle

```
Bundler could not find compatible versions for gem "railties":
  In Gemfile:
    coffee-rails (>= 4.2.1) was resolved to 4.2.1, which depends on
      railties (< 5.2.x, >= 4.0.0)
```

### System configuration
**Rails version**:
5.2.0-alpha
**Ruby version**:
2.4.1p111

**Comment by kaspth** [comment]
@dhh has been championing removing upper version restrictions and has shipped gem versions without those restrictions. So that's the long term answer.

But yeah, somebody needs to remove the upper bound on coffee-rails. Please do PR!

**Comment by dhh** [comment]
@rafaelfranca Can you cut a new gem for coffee-rails? I hit this issue too.

**Comment by dhh** [comment]
@kaspth The upper-bound was moved to Rails 6 in the referenced PR that was already merged. IMO, we shouldn't even have that bound in place, but if @rafaelfranca accepts responsibility for vetting and releasing in conjunction with 6.0, I'll leave that up to him.

**Comment by rafaelfranca** [comment]
Just released 4.2.2 and removed the the upper version restriction. Also gave access to the entire core team to release the gem.

**Comment by dhh** [comment]
🙏
